File: 0176_Problem_1.rs
// code_report Solution
// Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-176/problems/count-negative-numbers-in-a-sorted-matrix/
// Problem Link (Practice): https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
// Video Link:              https://youtu.be/pDbDtGn1PXk
// Rust Playground Link:    https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3641de477438f32ad2faa2ef544aaa15

pub fn count_negatives(grid: Vec<Vec<i32>>) -> i32 {
    return grid.iter()
               .flatten()
               .filter(|i| i < &&0i32)
               .count() as i32;
}

File: 0176_Problem_1.jl
# code_report Solution
# Problem Link (Contest):  https://leetcode.com/contest/weekly-contest-176/problems/count-negative-numbers-in-a-sorted-matrix/
# Problem Link (Practice): https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/

function countnegatives(grid)
    @pipe grid                |> 
        vec                   |> 
        filter(x -> x < 0, _) |> 
        length
end

# One-liner
function countnegatives(grid)
    @pipe grid |> vec |> filter(x -> x < 0, _) |> length
end

# Updated solution Aug 26, 2022
function countnegatives(grid)
    sum(grid .< 0)
end
